mesonbuild/mdist.py

```diff
@@ -149,112 +233,97 @@ def copy_git(self, src: T.Union[str, os.PathLike], distdir: str, revision: str =
         with tempfile.TemporaryFile() as f:
             subprocess.check_call(cmd, cwd=src, stdout=f)
             f.seek(0)
-            t = tarfile.open(fileobj=f) # [ignore encoding]
+            t = tarfile.open(fileobj=f)  # [ignore encoding]
             t.extractall(path=distdir)
 
-    def process_git_project(self, src_root: str, distdir: str) -> None:
-        if self.have_dirty_index():
-            handle_dirty_opt(msg_uncommitted_changes, self.options.allow_dirty)
-        if os.path.exists(distdir):
-            windows_proof_rmtree(distdir)
-        repo_root = self.git_root(src_root)
-        if repo_root.samefile(src_root):
-            os.makedirs(distdir)
-            self.copy_git(src_root, distdir)
-        else:
-            subdir = Path(src_root).relative_to(repo_root)
-            tmp_distdir = distdir + '-tmp'
-            if os.path.exists(tmp_distdir):
-                windows_proof_rmtree(tmp_distdir)
-            os.makedirs(tmp_distdir)
-            self.copy_git(repo_root, tmp_distdir, subdir=str(subdir))
-            Path(tmp_distdir, subdir).rename(distdir)
-            windows_proof_rmtree(tmp_distdir)
-        self.process_submodules(src_root, distdir)
-
     def process_submodules(self, src: str, distdir: str) -> None:
-        module_file = os.path.join(src, '.gitmodules')
-        if not os.path.exists(module_file):
+        module_file = Path(src) / '.gitmodules'
+        if not module_file.exists():
             return
```

Comment:
Unrelated to refactoring mdist for Mercurial, if this commit makes sense it should be in its own commit.

Reply:
For simplicity, I reverted these changes.